redbot/core/utils/views.py

```diff
@@ -431,3 +431,105 @@ async def auth_button(self, interaction: discord.Interaction, button: discord.Bu
         return await interaction.response.send_modal(
             SetApiModal(self.default_service, self.default_keys)
         )
+
+
+class ConfirmView(discord.ui.View):
+    """
+    A simple ``discord.ui.View`` used for confirming something.
+
+    Parameters
+    ----------
+    author: Optional[discord.abc.User]
+        The user who you want to be interacting with the confirmation.
+        If this is omitted anyone can click yes or no.
+    timeout: float
+        The timeout of the view in seconds. Defaults to ``180`` seconds.
+    disable_buttons: bool
+        Whether to disable the buttons after the timeout. Defaults to ``False``.
+
+    Examples
+    --------
+    Using the view::
+
+        view = ConfirmView(ctx.author)
+        view.message = await ctx.send("Are you sure you about that?", view=view)
+        # attach the message to the view this way to ensure clean removal
+        # when the view is finished.
+        await view.wait()
+        if view.result:
+            await ctx.send("Okay I will do that.")
+        else:
+            await ctx.send("I will not be doing that then.")
+
+    Disabling if the buttons are not pressed::
+
+        view = ConfirmView(ctx.author, disable_buttons=True)
+        view.message = await ctx.send("Are you sure you about that?", view=view)
+        await view.wait()
+        if view.result:
+            await ctx.send("Okay I will do that.")
+        else:
+            await ctx.send("I will not be doing that then.")
+
+    Attributes
+    ----------
+    result: Optional[bool]
+        The result of the confirm view.
+    author: Optional[discord.abc.User]
+        The author of the message who is allowed to press the buttons.
+    message: Optional[discord.Message]
+        The message the confirm view is sent on. This can be set while
+        sending the message. This can also be left as ``None`` in which case
+        if the view is never interacted with nothing will happen in ``on_timeout``
+    disable_buttons: bool
+        Whether to disable the buttons instead of remove the buttons if the message
+        has been seen by the view.
+    """
+
+    def __init__(
+        self,
+        author: Optional[discord.abc.User] = None,
+        *,
+        timeout: float = 180.0,
+        disable_buttons: bool = False,
+    ):
+        if timeout is None:
+            raise TypeError("This view should not be used as a persistent view.")
+        super().__init__(timeout=timeout)
+        self.result: Optional[bool] = None
+        self.author: Optional[discord.abc.User] = author
+        self.message: Optional[discord.Message] = None
+        self.disable_buttons = disable_buttons
+
+    async def on_timeout(self):
+        if self.message is None:
+            return
+            # we can't do anything here if message is none
```

Comment:
That's a weird order, comments generally are placed before the action they talk about:
```suggestion
            # we can't do anything here if message is none
            return
```